kiva/_marker_renderer.pyx

```diff
@@ -0,0 +1,126 @@
+
+        if _points.shape[1] != 2:
+            msg = "points argument must be an iterable of (x, y) pairs."
+            raise ValueError(msg)
+
+        self._this.draw_marker_at_points(
+            &_points[0][0],
+            _points.shape[0],
+            _size, _marker,
+            &_fill[0],
+            &_stroke[0]
+        )
+
+    def transform(self, sx, shy, shx, sy, tx, ty):
+        """transform(sx, shy, shx, sy, tx, ty)
+        Set the transform to be applied to the markers.
+
+        :param sx: Scale in X
+        :param shy: Shear in X
+        :param shx: Shear in Y
+        :param sy: Scale in Y
+        :param tx: Translation in X
+        :param ty: Translation in Y
+        """
+        cdef:
+            double _sx = <double>sx
+            double _shy = <double>shy
+            double _shx = <double>shx
+            double _sy = <double>sy
+            double _tx = <double>tx
+            double _ty = <double>ty
+
+        self._this.transform(_sx, _shy, _shx, _sy, _tx, _ty)
+
+
+cdef class MarkerRendererABGR32(MarkerRendererBase):
+    def __cinit__(self, uint8_t[:,:,::1] image, bottom_up=True):
+        self.base_init(image)
+        self._this = <renderer_base_t*> new renderer_abgr32_t(
+            &image[0][0][0], image.shape[1], image.shape[0], image.strides[0], bottom_up
+        )
+
+cdef class MarkerRendererARGB32(MarkerRendererBase):
+    def __cinit__(self, uint8_t[:,:,::1] image, bottom_up=True):
+        self.base_init(image)
+        self._this = <renderer_base_t*> new renderer_argb32_t(
+            &image[0][0][0], image.shape[1], image.shape[0], image.strides[0], bottom_up
+        )
+
+cdef class MarkerRendererBGRA32(MarkerRendererBase):
+    def __cinit__(self, uint8_t[:,:,::1] image, bottom_up=True):
+        self.base_init(image)
+        self._this = <renderer_base_t*> new renderer_bgra32_t(
+            &image[0][0][0], image.shape[1], image.shape[0], image.strides[0], bottom_up
+        )
+
+cdef class MarkerRendererRGBA32(MarkerRendererBase):
+    def __cinit__(self, uint8_t[:,:,::1] image, bottom_up=True):
+        self.base_init(image)
+        self._this = <renderer_base_t*> new renderer_rgba32_t(
+            &image[0][0][0], image.shape[1], image.shape[0], image.strides[0], bottom_up
+        )
+
+cdef class MarkerRendererBGR24(MarkerRendererBase):
+    def __cinit__(self, uint8_t[:,:,::1] image, bottom_up=True):
+        self.base_init(image)
+        self._this = <renderer_base_t*> new renderer_bgr24_t(
+            &image[0][0][0], image.shape[1], image.shape[0], image.strides[0], bottom_up
+        )
+
+cdef class MarkerRendererRGB24(MarkerRendererBase):
+    def __cinit__(self, uint8_t[:,:,::1] image, bottom_up=True):
+        self.base_init(image)
+        self._this = <renderer_base_t*> new renderer_rgb24_t(
+            &image[0][0][0], image.shape[1], image.shape[0], image.strides[0], bottom_up
+        )
```

Comment:
You think there's a nicer way to do these specializations?

Reply:
_has no understanding whatsoever about templates/specializations so cannot provide constructive comments_ Should we ask internally if anyone knows better?

Reply:
There are options: https://martinralbrecht.wordpress.com/2017/07/23/adventures-in-cython-templating/